File: Data_Structure/Python/LinearSearch.py
print("Enter the number of elements in the list:")
n = int(input())
print("Enter the elements of the list:")
elements = []
for i in range(n):
    elements.append(int(input()))
print("Enter the element to search for:")
target = int(input())

def linear_search(arr, target):
    for index, value in enumerate(arr):
        if value == target:
            return index
    return -1

result = linear_search(elements, target)
if result != -1:
    print("Element found at index:", result)
else:
    print("Element not found.")

File: Data_Structure/C/LinearSearch.c
#include <stdio.h>

int linearSearch(int arr[], int size, int target){
    for(int i =0; i< size; i++){
        if(arr[i] == target)
            return i;
    }
    return -1;
}

int main(){
    printf("Enter the size of the array: ");
    int size;
    scanf("%d", &size);
    int arr[size];

    printf("Enter the elements of the array:\n");
    for(int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }
    printf("Enter the target element to search: ");
    int target;
    scanf("%d", &target);
    int result = linearSearch(arr, size, target);
    if(result != -1){
        printf("Element found at index: %d\n", result);
    } else {
        printf("Element not found in the array.\n");   
    }
    return 0;
}